cmd/access/node_builder/access_node_builder.go

```diff
@@ -722,6 +723,15 @@ func (builder *FlowAccessNodeBuilder) BuildExecutionSyncComponents() *FlowAccess
 					if err != nil {
 						return nil, fmt.Errorf("could not load checkpoint file: %w", err)
 					}
+
+					// this is only supported when bootstrapping a new register db, otherwise the
+					// register db will throw exceptions when indexing an already indexed height.
+					if builder.forceResetIndexerHeight {
```

Comment:
shouldn't this actually always be set like that? is there a case where we would only remove the checkpoint content?

Reply:
currently, we can only remove the register data. events, results, etc, are stored in the protocol db, and can't be removed without some more code.

This requirement is here because there's a race condition between setting the register `latest` height and the indexer's current height, where it's possible the register height is updated first. In that case, data could be missed if someone rebooted with this flag enabled and an existing db.